src/utils/stac.ts

```diff
@@ -33,12 +33,19 @@ interface _Contact { // see: https://github.com/stac-extensions/contacts?tab=rea
 
 export type Contact = RequireAtLeastOne<_Contact, "name" | "organization">;
 
-export interface Dimension {
-  type: string;
-  axis?: "x" | "y" | "z";
-  description?: string;
-  unit?: string;
-}
+export type Dimension =
```

Comment:
I think it's a good idea to define the `Dimension` type more properly. But I'm wondering if we should take this moment to actually follow [the spec](https://github.com/stac-extensions/datacube?tab=readme-ov-file#dimension-object) more closely. I'd translate the spec such that all fields which are required don't get a `?` and all the others do get one.
But maybe it's also overkill for this PR.

Reply:
I think that all the members that we currently define do exactly that. Do you mean we should use the chance to also add the other optional members, even though we don't use them?

Reply:
I think `axis` only exists for `spatial`.